### DISCUSSION TITLE
Stack screens animation and Pan Responder

### DISCUSSION THREAD

**Discussion by Estebank94** [react-navigation/react-navigation#9711] [Q&A]
Hello,

Currently I am using `GestureRecognizer` from `react-native-swipe-gestures` to detect when the user swipes up and call the navigate function.
I was wondering if there is a way to animate the navigation of a stack screen as the user slides his finger across the screen. This way I could make the screen follow the user's finger vertically. I am guessing I could use Pan Responder but I didn't find any documentation about this.

Did anyone try making something like this?

**Comment by satya164** [comment]
https://reactnavigation.org/docs/stack-navigator/#gestureenabled
https://reactnavigation.org/docs/stack-navigator/#animation-related-options

**Reply by Estebank94** [reply]
Thanks, that's a bit useful but it doesn't have any example that uses Pan Gesture.

**Comment by Estebank94** [comment]
This is what I was able to do until now:

https://user-images.githubusercontent.com/5116089/124300292-cc760100-db34-11eb-855f-c73fdc98a244.mov

There are two main problems here: 
1. The animation is not smooth.
2. The previous screen disappears.

This is the code:

```javascript
const HomeFeed = createStackNavigator();

const HomeFeedStack = (props: any) => {
  const translateY = useRef(new Animated.Value(0)).current;

  const handlePan = (evt) => {
    const { nativeEvent } = evt;
    console.log(nativeEvent);
    translateY.setValue(Math.abs(nativeEvent.translationY));
    if (Math.abs(nativeEvent.translationY) > 10) {
      props.navigation.navigate(HOME_FEED_DETAILS);
    }
  };

  const handlePanEnd = (event) => {
    if (event.nativeEvent.state === State.END) {
      if (Math.abs(event.nativeEvent.translationY) > 200) {
        console.log('FINISH ANIMATIN');
        Animated.timing(translateY, {
          toValue: 812,
          duration: 500,
          useNativeDriver: true,
        }).start();
      } else {
        console.log('GO BACK____');
        Animated.timing(translateY, {
          toValue: 0,
          duration: 500,
          useNativeDriver: true,
        }).start(() => props.navigation.goBack());
      }
    }
  };

  const forSwipe = ({ layouts: { screen } }) => {
    return {
      containerStyle: {
        borderRadius: 50,
      },
      cardStyle: {
        transform: [
          {
            translateY: translateY.interpolate({
              inputRange: [0, 812],
              outputRange: [screen.height - 50, 0],
              extrapolate: 'clamp',
            }),
          },
        ],
      },
    };
  };

  return (
    <HomeFeed.Navigator mode="modal">
      <HomeFeed.Screen
        name={HOME_FEED}
        children={() => (
          <HomeFeedScreen
            handlePan={handlePan}
            handlePanEnd={handlePanEnd}
          />
        )}
        options={{ headerShown: false }}
      />
      <HomeFeed.Screen
        name={HOME_FEED_DETAILS}
        children={() => <View style={{ flex: 1, backgroundColor: 'blue' }} />}
        options={{ headerShown: false, cardStyleInterpolator: forSwipe }}
      />
    </HomeFeed.Navigator>
  );
};
```

**Reply by satya164** [reply]
Gestures are built-in and this automatically happens with the options I linked to. Not sure why you need to use another library.

**Reply by Estebank94** [reply]
Because I don't want to use swipe gesture, I want the screen to follow the finger as the user moves it from the bottom to the top of the screen.

**Reply by satya164** [reply]
I don't get why else would screen move with user's finger if not for swipe gesture, but regardless, just set a transparent background color for your screen in `cardStyle` and animate whatever component you want and with the library you want - which will get you your desired behavior.

**Reply by Estebank94** [reply]
To be more clear of what I want to achieve look at the last example of [react-native-pager-view](https://github.com/callstack/react-native-pager-view). 
I don't want the user to swipe and then call the navigate function with fixed animation,  that's the default modal animation on react navigation.
I want the user to swipe up and animate the screen as the users moves his finger, the screen should follow the finger. There are no examples on how to achieve that. 
I can achieve that with that library but then I am not really navigating between screens.

![RPReplay_Final1626203338](https://user-images.githubusercontent.com/5116089/125511791-710ea914-5430-4bcb-a1cf-c47629f34853.gif)

**Reply by satya164** [reply]
Use that library and wrap it in a custom navigator so that it integrates with React Navigation https://reactnavigation.org/docs/custom-navigators/

**Reply by Estebank94** [reply]
Thanks, that was a great idea. I did that and it worked

**Comment by Estebank94** [comment]
Done. Thanks @satya164
